src/main/java/seedu/address/logic/commands/DuplicateTransactionCommand.java

```diff
@@ -1,54 +1,82 @@
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TIMESTAMP;
 
 import java.util.List;
 
 import seedu.address.commons.core.index.Index;
 import seedu.address.commons.util.ToStringBuilder;
 import seedu.address.logic.Messages;
+import seedu.address.logic.commands.EditTransactionCommand.EditTransactionDescriptor;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
 import seedu.address.model.transaction.Transaction;
 
 /**
- * Duplicates a transaction identified using it's displayed index from the address book.
+ * Duplicates a transaction identified using its displayed index from the address book.
+ * Optional edits can be made to the duplicated transaction.
+ * Timestamp will be set to the current time if not specified.
  */
 public class DuplicateTransactionCommand extends Command {
 
     public static final String COMMAND_WORD = "duplicateTransaction";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-        + ": Duplicates the transaction identified by the index number used in the displayed transaction list.\n"
-        + "Parameters: INDEX (must be a positive integer)\n"
-        + "Example: " + COMMAND_WORD + " 1";
-
-    public static final String MESSAGE_DUPLICATE_TRANSACTION_SUCCESS = "Duplicated Transaction: %1$s";
+        + ": Duplicates the transaction identified by the index number with optional edits. "
+        + "Timestamp will be set to the current time if not specified.\n"
+        + "Parameters: INDEX (must be a positive integer) "
+        + "[" + PREFIX_COST + "COST] "
+        + "[" + PREFIX_DESCRIPTION + "DETAILS] "
+        + "[" + PREFIX_NAME + "PAYEE NAME] "
+        + "[" + PREFIX_TIMESTAMP + "TIMESTAMP]\n"
+        + "Example: " + COMMAND_WORD + " 1 "
+        + PREFIX_COST + "10.00 "
+        + PREFIX_DESCRIPTION + "Bought a book "
+        + PREFIX_NAME + "John Doe "
+        + PREFIX_TIMESTAMP + "2020-01-01 12:00";
+
+    public static final String MESSAGE_DUPLICATE_TRANSACTION_SUCCESS = "New duplicated transaction added: %1$s";
+
+    public static final String MESSAGE_TRANSACTION_NOT_RELEVANT =
+        "The duplicated transaction does not affect your balances";
 
     private final Index targetIndex;
+    private final EditTransactionDescriptor duplicateTransactionDescriptor;
 
-    public DuplicateTransactionCommand(Index targetIndex) {
+    /**
+     * @param targetIndex                    of the transaction in the filtered transaction list to duplicate
+     * @param duplicateTransactionDescriptor details to edit the duplicated transaction with
+     */
+    public DuplicateTransactionCommand(Index targetIndex, EditTransactionDescriptor duplicateTransactionDescriptor) {
         this.targetIndex = targetIndex;
+        this.duplicateTransactionDescriptor = duplicateTransactionDescriptor;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
```

Comment:
Just realised that we do not check if the edited/duplicated transaction has the same values as an existing transaction (i.e. same timestamp, description, payee, etc.) 

Given that we are making transactions unique, should we add checks for these in our execute command?

<img width="300" alt="Screenshot 2023-10-27 at 3 25 49 PM" src="https://github.com/AY2324S1-CS2103T-W17-3/tp/assets/74343245/ecd4d9c2-8289-4fec-b5e4-b855046a0743">

^ recreated with editTransactionCommand

Reply:
True, we should discuss this in the next meeting.